### PR TITLE
Add Nitrogen8mm MACHINE conf file and modify Nitrogen8m conf file

### PR DESCRIPTION
**Pull request by chrisdimich**

### CODE REVIEW

File: conf/machine/nitrogen8m.conf [hunk 1]
```diff
@@ -11,10 +11,15 @@ require conf/machine/include/arm/arch-arm64.inc
 
 # Kernel configuration
 PREFERRED_PROVIDER_virtual/kernel ??= "linux-boundary"
+PREFERRED_VERSION_linux-boundary = "4.14%"
```
Comment (otavio) [inline conf/machine/nitrogen8m.conf]: Why?

File: conf/machine/nitrogen8m.conf [hunk 2]
```diff
@@ -11,10 +11,15 @@ require conf/machine/include/arm/arch-arm64.inc
...
 KERNEL_DEVICETREE = "freescale/imx8mq-nitrogen8m.dtb \
 	freescale/imx8mq-nitrogen8m-m4.dtb \
+	freescale/imx8mq-nitrogen8m_som.dtb \
+	freescale/imx8mq-nitrogen8m_som-m4.dtb \
+	freescale/imx8mq-nitrogen8m-tc358743.dtb \
 "
 KERNEL_IMAGETYPE = "Image"
+KERNEL_DEFCONFIG = "boundary_defconfig"
```
Comment (otavio) [inline conf/machine/nitrogen8m.conf]: All this must be split on a specific commit.

File: conf/machine/nitrogen8mm.conf [hunk 3]
```diff
@@ -0,0 +1,53 @@
+#@TYPE: Machine
+#@NAME: Boundary Devices Nitrogen8M
+#@SOC: i.MX8 MQ
+#@DESCRIPTION: Machine configuration for Boundary Devices Nitrogen8M
+#@MAINTAINER: Chris Dimich <chris.dimich@boundarydevices.com>
+
+MACHINEOVERRIDES =. "mx8:mx8mm:mx8m:mx8mq:"
+
+require conf/machine/include/imx-base.inc
+require conf/machine/include/arm/arch-arm64.inc
```
Comment (otavio) [inline conf/machine/nitrogen8mm.conf]: Here as well. Another commit.

File: conf/machine/nitrogen8m.conf (hunk 1 above)
Comment (chrisdimich) [reply conf/machine/nitrogen8m.conf]: We have support for our 4.14 kernel in meta-boundary and we wish to AUTOREV and use our own boundary kernel defconfig.

**Comment by otavio** [comment]
@chrisdimich the revert is not required. In fact, it is wrong as it is reverting something added on this PR.

Please follow the OpenEmbedded commit guidelies[1].

1. https://www.openembedded.org/wiki/Commit_Patch_Message_Guidelines

**Comment by otavio** [comment]
@gibsson please assist @chrisdimich internally.

**Comment by chrisdimich** [comment]
Hi Otavio,

@otavio I spoke with Gary and we are going to do a new pull request, but we will have many more changes in addition to this.

**Comment by otavio** [comment]
Create small PRs so it is easier to review. Also, it must be against master.

**Pull request closed without merging**